### PR TITLE
Fix for #53: Refactored reference to deprecated method

### PR DESCRIPTION
**Pull request by lmarniazman**
Fix for #53

### CODE REVIEW

File: src/main/java/com/devonfw/ide/sonarqube/common/impl/config/ConfigurationFactory.java [hunk 1]
```diff
@@ -31,28 +33,29 @@ public ConfigurationFactory() {
   }
 
   /**
-   * @param fileToScan the {@link File} to analyze.
+   * @param inputFile the {@link InputFile} to analyze.
    * @return the {@link Configuration} responsible for the project owning the given {@link File}.
    */
-  public static Configuration get(File fileToScan) {
+  public static Configuration get(InputFile inputFile) {
```
Comment (hohwille) [inline src/main/java/com/devonfw/ide/sonarqube/common/impl/config/ConfigurationFactory.java]: Actually InputFile is a very specific/proprietary API of sonarqube. To finally make use of this you internally convert it to a path. Could you use `Path` already here and do the conversion to `Path` in `DevonArchitectureComponentCheck` to keep this class portable. We can then test it easily without the need to mock sonarqube `InputFile`.

Comment (lmarniazman) [reply src/main/java/com/devonfw/ide/sonarqube/common/impl/config/ConfigurationFactory.java]: The thing is that methods that return a `Path` are marked as deprecated as well. What I can do is get the path in a string via `context.getInputFile().uri().getPath()` and then just make the `public static Configuration get()` method take a String as parameter.

Comment (lmarniazman) [reply src/main/java/com/devonfw/ide/sonarqube/common/impl/config/ConfigurationFactory.java]: I refactored the cases where I used InputFile to String now and adjusted the code accordingly. Tests are fine and logs show the same output as before.

File: src/main/java/com/devonfw/ide/sonarqube/common/impl/config/ConfigurationFactory.java [hunk 2]
```diff
@@ -33,29 +31,30 @@ public ConfigurationFactory() {
   }
 
   /**
-   * @param inputFile the {@link InputFile} to analyze.
+   * @param path to the file to analyze.
    * @return the {@link Configuration} responsible for the project owning the given {@link File}.
    */
-  public static Configuration get(InputFile inputFile) {
+  public static Configuration get(String path) {
```
Comment (hohwille) [inline src/main/java/com/devonfw/ide/sonarqube/common/impl/config/ConfigurationFactory.java]: Sorry for being picky, but `String` is still the wrong choice for a file.
Either use `java.io.File` or `java.nio.file.Path` - both are fine and none of them is deprecated.
I have no clue which `Path` type you hit that should be deprecated.
If SonarQube refactored to allow any kind of URI/URL and not only a file path, you can also change to `URL` here and use `url.openStream()` to read the content, then use `context.getInputFile().uri().toURL()`.

Comment (lmarniazman) [reply src/main/java/com/devonfw/ide/sonarqube/common/impl/config/ConfigurationFactory.java]: What I meant was that methods from the JavaFileScannerContext that return File or Path are now deprecated.
I returned the ConfigurationFactory to its original state (taking a java.io.File as param) and in DevonArchitectureComponentCheck.java created a file out of context.getInputFile().toString().
Is this fine?

**Pull request merged**